### PR TITLE
feat(B-9.5): Implement Priority-based Filtering + Approval Threshold

### LINKED ISSUE #3918: feat(reviewer): B-9.5 Priority-based Filtering + Approval Threshold for MultiSpecialistReviewer
## Background

Based on architecture review of `MultiSpecialistReviewer` against Blueprint and Wish Pool v2, identified a gap in handling conflicting specialist opinions.

## Problem

Current `_deduplicate_findings()` only does simple deduplication (keep highest severity at same location). It lacks:
1. **De-confliction logic** for contradictory suggestions (e.g., Security says add checks, Performance says remove checks)
2. **Approval threshold** strategy (Security must pass vs Pythonic can be force-approved)

This can cause "infinite loop" scenarios where Coder keeps modifying code to satisfy one specialist while breaking another.

### Priority-based Filtering
- **Security findings**: Must be addressed (blocking)
- **Performance findings**: High priority (should address)
- **Architecture/Pythonic findings**: Optional (can be ignored after retry > 2)

### Approval Threshold Strategy
```python
def should_force_approve(findings: List[SpecialistFinding], retry_count: int) -> bool:
    # Security must always pass
    security_blockers = [f for f in findings if f.specialist == 'SECURITY' and f.severity in ('high', 'critical')]
    if security_blockers:
        return False
    
    # After 2 retries, allow force-approve for non-security issues
    if retry_count >= 2:
        return True
    
    return False
```

## Blueprint Alignment

- **Section 3.1 (Planner v3)**: Self-refinement + Debate Engine v2
- **Section 3.3 (Agent Catalog V2)**: Judge Agent for arbitration
- **Wish Pool v2 North Star**: "AI 自己審查、自己修復" requires handling conflicting reviews

## Related

- Parent: EPIC B (#2595)
- Related: F-5 Review Consolidation (to be created)
- Related: E+F+I closed loop (EPIC I #3342)

## Acceptance Criteria

- [ ] Priority-based filtering implemented in `_deduplicate_findings()`
- [ ] Approval threshold strategy with retry count awareness
- [ ] Security findings always blocking
- [ ] Non-security findings can be force-approved after retry threshold
- [ ] Telemetry for force-approve events

/cc @RC918

### PR DESCRIPTION
**Pull request by RC918**
# feat(B-9.5): Implement Priority-based Filtering + Approval Threshold

## Summary

Implements the core logic for B-9.5 Priority-based Filtering to prevent Coder Agent from entering infinite loops when specialists have conflicting opinions (e.g., Security says add checks, Performance says remove checks).

**Priority Rules Implemented:**
- Security findings (high/critical): ALWAYS blocking - never force-approve
- Performance findings: Require 3+ retries before force-approve
- Architecture/CORRECTNESS findings: Can be force-approved after 2 retries

**Changes:**
- `ForceApproveResult` dataclass for structured decision results with `to_dict()` for telemetry
- `should_force_approve(findings, retry_count)` - main decision function
- `filter_findings_by_priority(findings, retry_count)` - convenience function
- 19 comprehensive tests covering all priority rules and edge cases

## Review & Testing Checklist for Human

- [ ] **Verify priority rules match Issue #3918 spec** - Security high/critical always blocking, Performance needs 3+, Architecture/CORRECTNESS needs 2+
- [ ] **Note: Integration not included** - This PR adds the core logic only. A follow-up is needed to wire `should_force_approve()` into `langgraph_orchestrator.py` where `retry_count` is tracked
- [ ] **Telemetry not yet added** - The `to_dict()` method is ready for telemetry, but actual logging calls are not included
- [ ] **Empty findings behavior** - Returns `should_approve=False` with "No findings to evaluate" - verify this is the intended conservative default

**Test Plan:**
1. Run `pytest handoff/20250928/40_App/orchestrator/review_context/tests/test_force_approve.py -v` to verify all 19 tests pass
2. Review the priority rules in `should_force_approve()` against the spec in Issue #3918

### Notes

Fixes #3918

Blueprint Alignment:
- Section 3.1 (Planner v3 Self-refinement): Self-correction with limits
- Section 3.3 (Judge Agent): Arbitration for conflicting opinions

Link to Devin run: https://app.devin.ai/sessions/55832c623b294639871d23b39290eae5
Requested by: Ryan Chen (@RC918)

---

> [!NOTE]
> Introduces B-9.5 priority-based filtering and approval thresholds to prevent looped disagreements among specialists.
> 
> - Adds `ForceApproveResult` dataclass with `to_dict()` for structured decisions/telemetry
> - New `should_force_approve(findings, retry_count)` applying rules: `SECURITY` high/critical always block; `PERFORMANCE` needs 3+ retries; `ARCHITECTURE`/`CORRECTNESS` allowed after 2 retries
> - Adds `filter_findings_by_priority(findings, retry_count)` convenience helper returning remaining actionable findings
> - New tests `tests/test_force_approve.py` covering rules, mixed cases, empty input, and filtering behavior
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 591a61ca852d98377f0cf294742fe17e5d636646. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: handoff/20250928/40_App/orchestrator/review_context/multi_specialist_reviewer.py [hunk 1]
```diff
@@ -227,6 +227,182 @@ def get_weighted_findings(
...
+
+    # If retry_count >= 2 and no security blockers, we can force-approve
+    if retry_count >= 2:
+        return ForceApproveResult(
+            should_approve=True,
+            reason=f"Force-approved after {retry_count} retries (no security blockers)",
+            blocked_by=[],
+            filtered_findings=filtered,
+        )
+
```
Comment (RC918) [inline handoff/20250928/40_App/orchestrator/review_context/multi_specialist_reviewer.py]: IMPACT: This condition creates a logical inconsistency by allowing force-approval when retry_count >= 2, which contradicts the documented rule that PERFORMANCE findings require 3+ retries. A finding with only PERFORMANCE issues would be incorrectly approved at retry_count=2, violating the specified policy. The earlier check at line 331 already handles the performance threshold, making this blanket approval dangerous.

**Review by RC918 — COMMENTED** [review]
## MorningAI Code Review
